### ISSUE TITLE
Zeek unable to process egress traffic - Ethertype 0x8988 in ERSPAN

### ISSUE THREAD

**Issue by jmuellerlahn** [zeek/zeek#1928]
Hello,

we have a setup where we use ERSPAN and Zeek. The ERSPAN decapsulation works so well with Zeek for the ingress traffic, but not for egress traffic, so we only see half duplex in Zeek.

After analyzing the tcpdump and inspecting it with Wireshark, we found out, that the error is based upon the Ethertype 0x8988. Wireshark with the Version 3.0.2 couldn't handle the traffic either, after upgrading Wireshark to version 3.6.1 we could see ingress and egress traffic.

Also there is a bug entry at Cisco:

**Symptom:**
Fabric egress packets that are ERSPAN'd still have the PTP header. Wireshark might not be able to decode the packets, and instead shows frames with ethertype 0x8988.
**Conditions:**
This issue occurs with switches that have -EX or a later extension in the PID, and PTP is enabled in the fabric.
**Workaround:**
Wireshark supports the ability to decode PTP headers. To do this right-click the ethertype value in wireshark and select "decode as". Set ethertype 0x8988 to decode as "Cisco TTag". The contents of the packet should then be viewable in wireshark.
Note, if "Cisco TTag" isn't an option make sure that wireshark is updated.

Can you confirm and fix that the issue exists in Zeek.

Thanks in advance,

Johannes

**Comment by 0xxon** [comment]
Hi Johannes,

I assume that you are correct and that we will currently not parse traffic that has this additional header and shows ethertype 0x8988 - traffic with unknown ethertypes is currently not parsed by Zeek.

Do you happen to have a trace file with  some traffic that is affected? That would help a lot with potentially implementing support for this.

Johanna

**Comment by jmuellerlahn** [comment]
Hello Johanna,

thanks for your fast reply. Sure I can handle you over some traffic files. Is there a secure way to sent you those files, as I don't want to have them in the public.

Regards,

Johannes

**Comment by jmuellerlahn** [comment]
Hello Johanna,

now you have some files that are not marked confidential: [erspan_pcaps.zip](https://github.com/zeek/zeek/files/7847782/erspan_pcaps.zip)

 Hope you can work with them.

Thanks in advance.

Regards,



Johannes

**Comment by jmuellerlahn** [comment]
Maybe I have to mention the we use Zeek Its version: zeekversion = 3.0.12

**Comment by jmuellerlahn** [comment]
Hello Johanna,

I tried to understand the problem a little bit more. I asked myself, what has been fixed in Wireshark 3.6.1 that it could handle the Ethertype 0x8988, so I found this sourcecode: https://github.com/wireshark/wireshark/blob/master/epan/dissectors/packet-pa-hbbackup.c

This is interesting: 

/* 2do:
 * - Find out the meaning of the 6 bytes header: timestamp?
 * - Handle trailer bytes correctly
 */

I'm not an C++ Expert, but maybe this helps you to fix the issue.

Thanks.

Johannes

**Comment by timwoj** [comment]
Adding support for new ethertypes shouldn't be too hard for GRE/ERSPAN. That code is all in https://github.com/zeek/zeek/blob/master/src/packet_analysis/protocol/gre/GRE.cc. Thanks for the pcaps, I'll take a look.

Just a heads up that any fixes we do here would be for later versions of Zeek. 3.0 is an old LTS version, superceded by 4.0 at this point. We could potentially pull this fix into a patch release for 4.0 as well, if needed.

**Comment by timwoj** [comment]
Now that I have a minute to look at the packet traffic, this would probably require a new packet analyzer to implement properly. The Ethernet analyzer has specific forwarding rules for various Ethernet types, and this one isn't listed. You can *sort of* get away with using the skip analyzer here, by doing something like this:

```
export {
        redef PacketAnalyzer::SKIP::skip_bytes = 8;
}

event zeek_init()
{
	PacketAnalyzer::register_packet_analyzer(PacketAnalyzer::ANALYZER_ETHERNET, 0x8988, PacketAnalyzer::ANALYZER_SKIP);
}
```

Note though that `skip_bytes` is a global value and you can only set it to one value for the life of the skip analyzer (i.e. the life of the zeek process). The skip analyzer defaults to trying to forward into the IP analyzer, so as long as the only thing that could possibly come after the PAHB header is an IP header, you might be ok. It's not looking at the actual data in the PAHB header though. It simply skips 8 bytes in the buffer, and tries to process the next header as IP.
